### PR TITLE
PROTOTECH-14: multiply t0debt with inflator when calculating meaningful deposit

### PR DESCRIPTION
**Pull request by grandizzy**
# Description of change
## High level
* Correction to Meaningful Deposit for In-Auction Debt Neglects Inflator
  * multiply t0 debt in auction with inflator when correcting meaningful deposit
  * see https://github.com/Fixed-Point-Solutions/prototech-ajna-audit/issues/14

# Description of bug or vulnerability and solution
* Meaningful deposit wrongly corrected
* change `PoolCommons._meaningfulDeposit` logic  from
```solidity
        meaningfulDeposit_ -= Maths.min(meaningfulDeposit_, t0DebtInAuction_);
```
to
```solidity
        meaningfulDeposit_ -= Maths.min(
            meaningfulDeposit_,
            Maths.wmul(t0DebtInAuction_, inflator_)
        );
 ```
* added unit tests for `PoolCommons._meaningfulDeposit` branches
* fixed baseline tests

# Contract size
## Pre Change
```
  PoolCommons              -   9,443B  (38.42%)
```
## Post Change
```
  PoolCommons              -   9,447B  (38.44%)
```

# Gas usage
## Pre Change
```
| Function Name                        | min             | avg    | median | max     | # calls |
| addQuoteToken                        | 119822          | 176971 | 161647 | 711476  | 60004   |
| bucketTake                           | 311897          | 330272 | 330272 | 348648  | 4       |
| drawDebt                             | 247478          | 292168 | 266312 | 802543  | 136003  |
| kick                                 | 151711          | 221290 | 219662 | 1031621 | 48000   |
| kickWithDeposit                      | 195774          | 273194 | 269338 | 996200  | 8000    |
| moveQuoteToken                       | 267290          | 267290 | 267290 | 267290  | 1       |
| removeQuoteToken                     | 142805          | 163119 | 167036 | 188398  | 4000    |
| repayDebt                            | 574176          | 606318 | 596214 | 754485  | 16002   |
| settle                               | 348410          | 348410 | 348410 | 348410  | 1       |
| take                                 | 80551           | 81966  | 81656  | 313032  | 15998   |
```
## Post Change
```
| Function Name                        | min             | avg    | median | max     | # calls |
| addQuoteToken                        | 120153          | 177302 | 161978 | 711807  | 60004   |
| bucketTake                           | 308418          | 326793 | 326793 | 345169  | 4       |
| drawDebt                             | 247809          | 292499 | 266643 | 802874  | 136003  |
| kick                                 | 147225          | 221624 | 220000 | 1031952 | 48000   |
| kickWithDeposit                      | 196105          | 273535 | 269686 | 996531  | 8000    |
| moveQuoteToken                       | 277229          | 277229 | 277229 | 277229  | 1       |
| removeQuoteToken                     | 143070          | 163417 | 167279 | 188553  | 4000    |
| repayDebt                            | 574507          | 606649 | 596545 | 754816  | 16002   |
| settle                               | 348675          | 348675 | 348675 | 348675  | 1       |
| take                                 | 80802           | 82228  | 81920  | 313283  | 15998   |
```

### CODE REVIEW

File: tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol [hunk 1]
```diff
@@ -884,12 +884,12 @@ contract ERC721PoolCollateralTest is ERC721HelperContract {
                 pledgedCollateral:    1.139736976079754220 * 1e18,
                 encumberedCollateral: 4404023621.084799631606156246 * 1e18,
                 poolDebt:             439.681348088864224700 * 1e18,
-                actualUtilization:    0.061025469820976144 * 1e18,
+                actualUtilization:    0.750721153846153847 * 1e18,
```
Comment (grandizzy) [inline tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol:887]: big move, I think good sign, @mattcushman can you pls review?

File: tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol [hunk 2]
```diff
@@ -1165,16 +1165,16 @@ contract ERC721PoolCollateralTest is ERC721HelperContract {
             PoolParams({
                 htp:                  0,
                 lup:                  MAX_PRICE,
-                poolSize:             50.000004575591110007 * 1e18,
+                poolSize:             50.000004575591109992 * 1e18,
                 pledgedCollateral:    0,
                 encumberedCollateral: 0,
                 poolDebt:             0,
-                actualUtilization:    0.001552228747991919 * 1e18,
+                actualUtilization:    0.750721153846153847 * 1e18,
```
Comment (grandizzy) [inline tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol:1172]: same @mattcushman

File: tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol [hunk 3]
```diff
@@ -1165,16 +1165,16 @@ contract ERC721PoolCollateralTest is ERC721HelperContract {
...
                 targetUtilization:    0.328577182109433013 * 1e18,
                 minDebtAmount:        0,
                 loans:                0,
                 maxBorrower:          address(0),
-                interestRate:         0.04455 * 1e18,
+                interestRate:         0.05445 * 1e18,
```
Comment (grandizzy) [inline tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol:1177]: interest rate raise

File: tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol [hunk 4]
```diff
@@ -884,12 +884,12 @@ contract ERC721PoolCollateralTest is ERC721HelperContract {
...
                 encumberedCollateral: 4404023621.084799631606156246 * 1e18,
                 poolDebt:             439.681348088864224700 * 1e18,
-                actualUtilization:    0.061025469820976144 * 1e18,
+                actualUtilization:    0.750721153846153847 * 1e18,
                 targetUtilization:    0.328577182109433013 * 1e18,
                 minDebtAmount:        0,
                 loans:                0,
                 maxBorrower:          address(0),
-                interestRate:         0.0495 * 1e18,
+                interestRate:         0.0605 * 1e18,
```
Comment (grandizzy) [inline tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol:892]: increase of interest rate

File: tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol (hunk 1 above)
Comment (mattcushman) [reply tests/forge/unit/ERC721Pool/ERC721PoolCollateral.t.sol:887]: yes i think this is better...

**Review by mattcushman — APPROVED** [review]
lgtm

**Review by kmbarry1 — APPROVED** [review]
LGTM

Code change looks correct and test value changes are generally in the directions I expect (higher actual utilization, higher interest rates).

**Review by prat-gpt — APPROVED** [review]
LGTM!

**Pull request merged**